### PR TITLE
[CUBRIDMAN-5] CUBRID Java Stored Procedure Server Utility

### PR DESCRIPTION
**Pull request by hgryoo**
http://jira.cubrid.org/browse/CUBRIDMAN-5

- Add new section for Java Stored Procedure Server
- Revise existing sections related to the Java SP Server

### To Revise exsiting sections
- Introduction to CUBRID (intro.rst)
   - System Architecture
      - Process Structure
         - [ ] Revised figure (**will be considered at another issue**)
         - [x] Add a description about Java SP server roughly
   - CUBRID Characteristics
      - Java Stored Procedure
         - [x] Revise steps to execute Java stored procedure/function
         - [x] Add a link to a new section for javasp utility
- Installing and Upgrading
   - Configuring Environment Variables (env.rst)
      - OS Environment and Java Environment Variables
         - [x] Add a description about `JVM_PATH`
      - Port Setting
         - [x] Add contents for cub_javasp: `Add "%CUBRID%\bin\cub_javasp.exe” to open all ports for the CUBRID Java Stored Procedure server.`
         - [x] Add a row for javasp in the table
         - [x] Add port for javasp at system parameter info (CBRD-23626)
         - [ ] Add the relation of connection between CAS, cub_server, cub_javasp (**I'm not sure we need this**)
     - Upgrade
        - [ ] Cautions during upgrade (**It should be added in the PR for Upgrade, not in this PR**)
- CUBRID SQL
   - Java Stored Function/Procedure (jsp.rst) : Rewrite the structure of this section if required
        - [x] Environment Configuration for Java Stored Function/Procedure (moved)
        - [x] How to Write Java Stored Function/Procedure (no revision needed)
        - [x] Using Server-side Internal JDBC Driver (no revision needed)
        - [x] Connecting to Other Database (no revision needed)
        - [x] loadjava Utility (no revision needed)
        - [x] Loaded Java Class Publish (no revision needed)
        
- CUBRID Management
   - [x] Revise introduction
   - [x] CUBRID Services
   - [x] System Parameters (linked from the new section, Java SP Server System Parameters)

### To Add a new section
- CUBRID Management - Java Stored Procedure Server (control.rst)
  - [x] Starting Java SP Server
  - [x] Stopping Java SP Server
  - [x] Restarting Java SP Server
  - [x] Checking Java SP Status
  - [x] Configuring for CUBRID Java SP Server
          - Environment Configuration for Java Stored Function/Procedure (moved from CUBRID SQL)
          - Java SP Server System Parameters
          - Registering Java SP Server to cubrid service
  - [x] Java SP Server Log
     - [x] Error log
     - [x] Java log
  - [x] Java SP Server Errors

### CODE REVIEW

File: en/admin/index.rst [hunk 1]
```diff
@@ -42,6 +42,10 @@ The service management utilities are as follows:
 
     *   :ref:`cubrid heartbeat <cubrid-heartbeat>`
 
+*   Java SP Server utility : Operates and manages the process of the Java stored procedure (Java SP) server.
```
Comment (charischau) [inline en/admin/index.rst:45]: Operates --> operates 

(Capitalization After Colons: 
- When a colon introduces a list of of things, do not capitalize the first word after the colon unless it is a proper noun.
- When a colon introduces a phrase or an incomplete sentence that is meant to add information to the sentence before it, do not capitalize the first word after the colon unless it is a proper noun.

Therefore, there 'Operates' should change to 'operates', the lower case)

File: en/sql/jsp.rst [hunk 2]
```diff
@@ -21,77 +21,35 @@ The advantages of using Java stored functions/procedures are as follows:
...
-.. _jsp-environment-configuration:
+.. _jsp-starting-javasp:
 
-Environment Configuration for Java Stored Function/Procedure
-============================================================
+Starting Java Stored Procedure Server
+=====================================
 
-To use Java-stored functions/procedures in CUBRID, you must have JRE (Java Runtime Environment) 1.6 or better installed in the environment where the CUBRID server is installed. You can download JRE from the Developer Resources for Java Technology (`https://www.oracle.com/java/technologies <https://www.oracle.com/java/technologies>`_).
+You need to start a Java Stored Procedure server (Java SP server) for a database you want to use Java-stored procedures/functions.
```
Comment (charischau) [inline en/sql/jsp.rst]: for a database --> for the database

File: en/admin/control.rst [hunk 3]
```diff
@@ -2142,3 +2162,358 @@ The following shows how to use the CUBRID Manager (hereafter, CM) Administrator
...
+
+::
+
+    % cubrid javasp stop demodb
+
+    @ cubrid javasp stop: demodb
+    ++ cubrid javasp stop: success
+
+The following message is returned when CUBRID Java SP server has stopped. 
+
```
Comment (beyondykk9) [inline en/admin/control.rst:2209]: propose: The following message is returned when CUBRID Java SP server has been already stopped.

File: en/admin/control.rst [hunk 4]
```diff
@@ -2142,3 +2162,358 @@ The following shows how to use the CUBRID Manager (hereafter, CM) Administrator
...
+Configuring for CUBRID Java SP Server
+-------------------------------------
+
+.. _cubrid-javasp-environment-configuration:
+
+Environment Configuration for Java Stored Function/Procedure
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To use Java-stored functions/procedures in CUBRID, you must have JRE (Java Runtime Environment) 1.6 or better installed in the environment where the CUBRID server is installed. You can download JRE from the Developer Resources for Java Technology (`https://www.oracle.com/java/technologies <https://www.oracle.com/java/technologies>`_).
```
Comment (beyondykk9) [inline en/admin/control.rst]: propose: better --> later
for version, commonly use "later"

File: en/admin/control.rst (hunk 3 above)
Comment (charischau) [reply en/admin/control.rst:2209]: has been already stopped. --> has been stopped already.  
 might be more fluent.

File: en/admin/control.rst [hunk 5]
```diff
@@ -2142,3 +2162,358 @@ The following shows how to use the CUBRID Manager (hereafter, CM) Administrator
...
+    % export LD_LIBRARY_PATH
+
+*   An example of installing 32 Bit JDK 1.6 and configuring the environment variables in a bash shell ::
+
+    % JAVA_HOME=/usr/java/jdk1.6.0_10
+    % LD_LIBRARY_PATH=$JAVA_HOME/jre/lib/i386/:$JAVA_HOME/jre/lib/i386/client:$LD_LIBRARY_PATH
+    % export JAVA_HOME
+    % export LD_LIBRARY_PATH
+
+*   An example of installing 64 Bit JDK 1.6 and configuring the environment variables in a csh ::
```
Comment (beyondykk9) [inline en/admin/control.rst]: propoe: in a csh --> in a csh shell, similar with others

File: en/admin/control.rst [hunk 6]
```diff
@@ -2142,3 +2162,358 @@ The following shows how to use the CUBRID Manager (hereafter, CM) Administrator
...
+::
+
+    % cubrid service start
+    
+    @ cubrid master start
+    ++ cubrid master start: success
+    @ cubrid server start: demodb
+
+    This may take a long time depending on the amount of restore works to do.
+    CUBRID 10.2
```
Comment (beyondykk9) [inline en/admin/control.rst:2409]: 10.2 is right?

File: en/admin/control.rst [hunk 7]
```diff
@@ -2142,3 +2162,358 @@ The following shows how to use the CUBRID Manager (hereafter, CM) Administrator
...
+
+    ++ cubrid server start: success
+    @ cubrid server start: testdb
+
+    This may take a long time depending on the amount of recovery works to do.
+    CUBRID 10.2
```
Comment (beyondykk9) [inline en/admin/control.rst:2415]: 10.2 is right?

File: en/admin/control.rst [hunk 8]
```diff
@@ -2142,3 +2162,358 @@ The following shows how to use the CUBRID Manager (hereafter, CM) Administrator
...
+    ++ cubrid javasp start: success
+    @ cubrid broker start
+    ++ cubrid broker start: success
+
+.. _cubrid-javasp-server-log:
+
+CUBRID Java SP Server Log
+-------------------------
+
+The logs of CUBRID Java SP server are stored in the log/ directory under the installation directory. There are two types of log files depending on server process of CUBRID Java Stored Procedure Server.
```
Comment (beyondykk9) [inline en/admin/control.rst]: propose: two types of --> two kinds of

File: en/admin/control.rst [hunk 9]
```diff
@@ -2142,3 +2162,358 @@ The following shows how to use the CUBRID Manager (hereafter, CM) Administrator
...
++-------+----------------------------------+-----------------------------------------------------+-----------------------------------------------------------------------------------+
+| -901  | Java VM can not be started: ?    | Unexpected internal error occured in JVM library.   | Try installing the JRE again. If you keep getting the error,                      |
+|       |                                  | The JVM library may be broken, or there may be a    | try installing a different version of the JRE.                                    |
+|       |                                  | problem with the $CUBRID/java/jspserver.jar file.   | Try replacing it with the same CUBRID version of $CUBRID/java/jspserver.jar file. |
++-------+----------------------------------+-----------------------------------------------------+-----------------------------------------------------------------------------------+
+
+The following are error messages about the errors which can be occrured when there is a problem with the connection to Java SP server including it is not started.
```
Comment (beyondykk9) [inline en/admin/control.rst]: The following --> The followings

File: en/admin/index.rst [hunk 10]
```diff
@@ -8,7 +8,7 @@ CUBRID Management
 
 This chapter describes how the database administrators (**DBA**) operates the CUBRID system.
 
-*   It includes instructions on how to use the **cubrid** utility, which starts and stops various processes of the CUBRID server, the broker and manager server. See :doc:`/admin/control`.
+*   It includes instructions on how to use the **cubrid** utility, which starts and stops various processes of the CUBRID server, the broker, java stored procedure server and manager server. See :doc:`/admin/control`.
```
Comment (beyondykk9) [inline en/admin/index.rst]: the CUBRID server, the broker, java stored procedure server and manager server
-->
the CUBRID server, the broker, java stored procedure server, and manager server
(comma missed)

File: en/env.rst [hunk 11]
```diff
@@ -378,3 +383,18 @@ The following table summarizes the ports, based on the listening processes, used
...
+
+The following table summarizes the ports, based on the listening processes, used for CUBRID Java Stored Procedure (Java SP) server. The ports are identical regardless of the OS type.
+
++---------------+--------------+----------------------------+--------------------------+
+| Listener      | Requester    | Port                       | Firewall Port Setting    |
++===============+==============+============================+==========================+
+| cub_javasp    | CAS          | java_stored_procedure_port | Open                     |
++---------------+--------------+----------------------------+--------------------------+
+
+*   The port used when the CAS relays between Java SP server (cub_javasp) and cub_server, which CAS receives a call of the java stored procedure from cub_server and then CAS passed the call to the CUBRID Java SP server process through **java_stored_procedure_port** of cubrid.conf.
```
Comment (beyondykk9) [inline en/env.rst]: propose: The port used --> The port is used

File: en/admin/control.rst (hunk 9 above)
Comment (hgryoo) [reply en/admin/control.rst]: As far as I know, "The following" is used to introduce a list, a quotation, etc and it remains the same for both singular and plural.

File: en/admin/control.rst (hunk 7 above)
Comment (hgryoo) [reply en/admin/control.rst:2415]: It appears in all parts of the current manual. Updating the version in the examples needs to be done on a separate issue at once.

File: en/admin/control.rst (hunk 6 above)
Comment (hgryoo) [reply en/admin/control.rst:2409]: It appears in all parts of the current manual. Updating the version in the examples needs to be done on a separate issue at once.

File: en/intro.rst [hunk 12]
```diff
@@ -29,6 +29,8 @@ CUBRID is an object-relational database management system (DBMS) consisting of t
 
 *   The CUBRID Manager is a GUI tool that allows users to remotely manage the database and the broker. It also provides the Query Editor, a convenient tool that allows users to execute SQL queries on the database server. 
 
+*   The CUBRID Java SP server is an execuction server that processes Java stored procedures/functions passed from the database server.
```
Comment (beyondykk9) [inline en/intro.rst]: "passed" looks not necessary and "execuction" is misspelled
propose:
The CUBRID Java SP server is an execution server that processes Java stored procedures/functions from the database server.

File: en/sql/jsp.rst (hunk 2 above)
Comment (hgryoo) [reply en/sql/jsp.rst]: Thank you, I'll fix it

File: en/admin/control.rst (hunk 3 above)
Comment (hgryoo) [reply en/admin/control.rst:2209]: Thank you, I'll fix it

File: en/admin/control.rst (hunk 4 above)
Comment (hgryoo) [reply en/admin/control.rst]: Thank you, I'll fix it

File: en/admin/index.rst (hunk 10 above)
Comment (hgryoo) [reply en/admin/index.rst]: Thank you, I'll fix it

File: en/env.rst (hunk 11 above)
Comment (hgryoo) [reply en/env.rst]: Thank you, I'll fix it

File: en/intro.rst (hunk 12 above)
Comment (hgryoo) [reply en/intro.rst]: I'll fix the typo. thank you.
But if "passed" is removed, the meaning is changed. the "passed" is to represent the meaning that the database server sends a request to perform the stored procedure to the Java SP server. just starting with "from" may sound ambiguous.

File: en/admin/control.rst (hunk 8 above)
Comment (hgryoo) [reply en/admin/control.rst]: I revised the sentence as below.
```
The logs of CUBRID Java SP server are stored in the log/ directory under the installation directory. 
The following log files are created for CUBRID Java Stored Procedure Server per database.
*   Error Log ($CUBRID/log/[db_name]_java.err)
*   Java Log ($CUBRID/log/[db_name]_java.log)
```

File: en/intro.rst (hunk 12 above)
Comment (shparkcubrid) [reply en/intro.rst]: how about 'requested'? 
.... server that processes Java stored procedures/functions requested from the database server.

File: en/admin/control.rst (hunk 5 above)
Comment (hgryoo) [reply en/admin/control.rst]: Thank you. I appended 'shell'. but 'similar with others' seems redundant.

File: en/intro.rst (hunk 12 above)
Comment (hgryoo) [reply en/intro.rst]: I'll revise it to 'requested'. Thank you.

File: en/admin/control.rst [hunk 13]
```diff
@@ -2142,3 +2162,358 @@ The following shows how to use the CUBRID Manager (hereafter, CM) Administrator
...
+
+.. note::
+
+    For more details on what errors can be occured, see :ref:`cubrid-javasp-server-errors`.
+
+
+Java Log
+^^^^^^^^^
+
+An Java log of the JVM in the Java SP server is saved into **$CUBRID/log** directory, and and the format of the file name is *<db_name>_java.log*. The extension is ".log".
```
Comment (shparkcubrid) [inline en/admin/control.rst]: and and the format ==> and the...

Comment (hgryoo) [reply en/admin/control.rst]: Thank you.

**Review by shparkcubrid — APPROVED** [review]

**Review by beyondykk9 — APPROVED** [review]

**Review by charischau — APPROVED** [review]

**Pull request merged**